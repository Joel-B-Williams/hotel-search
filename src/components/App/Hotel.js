import React, { Component } from 'react';

class Hotel extends Component {

  defaultImage = (ev) => {
    ev.target.src = "https://i.imgur.com/qUvaEAl.png"
  }

  render () {
    return (
      <div className="hotel-list">
        <img src={this.props.hotelStaticContent.mainImage.url} onError={this.defaultImage} className="photo"/>
        <div className="details">
          <div className="hotel-name">
            {this.props.hotelStaticContent.name || "No name found" }
          </div>
          <div className="hotel-location">
            {this.props.hotelStaticContent.neighborhoodName || "Location unknown"}
          </div>
        </div>
        <div className="price">
          <table className="price-table">
            <tbody>
              <tr>
                <td>Per night: </td>
                <td className="price-per-night">
                  {/* find less risky method if possible */}
                  <p dangerouslySetInnerHTML={{__html: this.props.lowestAveragePrice.symbol}}></p>
                  {this.props.lowestAveragePrice.amount || "$$"}
                </td>
              </tr>
              <tr className="points-row">
                <td>Points earned:</td>
                <td className="points-earned">
                  {this.props.rewards.miles || "Reward unavailable"}
                </td>
              </tr>
            </tbody>
          </table>
          <button className="btn hotel-btn">Select Now</button>
        </div>
      </div>
    );
  }
}
  
export default Hotel;