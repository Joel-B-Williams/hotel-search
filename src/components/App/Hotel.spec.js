import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Hotel from './Hotel';

describe('Hotel', () => {
    const hotel = {
      "rewards":{"miles":8000},"lowestAveragePrice":{"currency":"USD","symbol":"&#36;","amount":234.0},"hotelStaticContent":{"mainImage":{"category":"EXTERIOR","url":"http://d2whcypojkzby.cloudfront.net/imageRepo/6/0/96/743/239/chidt-exterior-0227-hor-clsc_S.jpg"},"name":"Chicago Marriott Downtown Magnificent Mile","neighborhoodName":"Magnificent Mile"}}
    
    const wrapper = shallow(<Hotel 
        hotelStaticContent={hotel.hotelStaticContent}
        lowestAveragePrice={hotel.lowestAveragePrice}
        rewards={hotel.rewards} />);

    it('renders the component', () => {
      expect(wrapper.find('.hotel-list').exists()).toBe(true);
    });

    it('matches snapshot', () => {
      const comp = create(<Hotel 
        hotelStaticContent={hotel.hotelStaticContent}
        lowestAveragePrice={hotel.lowestAveragePrice}
        rewards={hotel.rewards} />);

      expect(comp.toJSON()).toMatchSnapshot();
    });
});