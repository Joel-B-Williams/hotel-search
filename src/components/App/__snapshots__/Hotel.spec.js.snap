// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotel matches snapshot 1`] = `
<div
  className="hotel-list"
>
  <img
    className="photo"
    onError={[Function]}
    src="http://d2whcypojkzby.cloudfront.net/imageRepo/6/0/96/743/239/chidt-exterior-0227-hor-clsc_S.jpg"
  />
  <div
    className="details"
  >
    <div
      className="hotel-name"
    >
      Chicago Marriott Downtown Magnificent Mile
    </div>
    <div
      className="hotel-location"
    >
      Magnificent Mile
    </div>
  </div>
  <div
    className="price"
  >
    <table
      className="price-table"
    >
      <tbody>
        <tr>
          <td>
            Per night: 
          </td>
          <td
            className="price-per-night"
          >
            <p
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#36;",
                }
              }
            />
            234
          </td>
        </tr>
        <tr
          className="points-row"
        >
          <td>
            Points earned:
          </td>
          <td
            className="points-earned"
          >
            8000
          </td>
        </tr>
      </tbody>
    </table>
    <button
      className="btn hotel-btn"
    >
      Select Now
    </button>
  </div>
</div>
`;
